File: Fibonacci.java
import java.util.Scanner;
import java.math.BigInteger;

public class Fibonacci {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        System.out.print("Enter the number of terms: ");
        int n = scanner.nextInt();
        
        // Original: int t1 = 0, t2 = 1;
        // Java's int type is a 32-bit signed integer with max value 2,147,483,647
        // Fibonacci numbers exceed this around F(47), causing overflow
        // BigInteger class provides arbitrary precision for large values
        BigInteger t1 = BigInteger.ZERO; // Using BigInteger for arbitrary precision
        BigInteger t2 = BigInteger.ONE;
        System.out.print("Fibonacci Series: ");
        
        for (int i = 1; i <= n; i++) {
            System.out.print(t1 + ", ");
            BigInteger nextTerm = t1.add(t2); // BigInteger uses .add() method
            t1 = t2;
            t2 = nextTerm;
        }
        
        System.out.println();
        scanner.close();
    }
}

File: Fibonacci.ts
import * as readline from 'readline';

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

rl.question('Enter the number of terms: ', (n: string) => {
    const terms: number = parseInt(n);
    // Original: let t1: number = 0; let t2: number = 1;
    // TypeScript's number type (same as JavaScript) uses 64-bit floating point
    // Limited to ~15-16 digits precision, causing errors for large Fibonacci numbers
    // bigint type provides arbitrary precision for accurate calculations
    let t1: bigint = 0n; // Using bigint (note the 'n' suffix)
    let t2: bigint = 1n;
    let result: string = "Fibonacci Series: ";
    
    for (let i: number = 1; i <= terms; i++) {
        result += t1 + ", ";
        let nextTerm: bigint = t1 + t2;
        t1 = t2;
        t2 = nextTerm;
    }
    
    console.log(result);
    rl.close();
});
